Cargo.toml

```diff
@@ -110,6 +111,18 @@ zstd = { version = "*", features = [
 async-compat = { version = "0.2.2", optional = true }
 async-stream = { version = "0.3.5", optional = true }
 wasm-streams = { version = "0.3.0", optional = true }
+object_store = { version = "0.8.0", default-features = false, features = [] }
+async-trait = "0.1.74"
+tokio = { version = "1.34.0", default-features = false }
+itertools = "0.12.0"
+url = "2.5.0"
+chrono = { version = "0.4.31", default-features = false, features = [ "wasmbind" ] }
+snafu = "0.7.5"
+async-std = "1.12.0"
+backon = "0.4.1"
+instant = { version = "0.1.12", features = ["wasm-bindgen"] }
```

Comment:
that is a _lot_ of deps... do you know how much the bundle size changes by?

Reply:
Right now it's sitting at 5.20MB, so about 100kB relative to main (5.09MB).

snafu, async-std and itertools could probably go, though because the latter two are unused, they're likely not contributing to the bundle.

Reply:
ah interesting; that's less of an addition than I expected (brotli-encoding that should bring it down to about 1MB; try `brotli -Z`). What's the compressed delta?

Reply:
Interesting - 1.22MB up from 1.15MB (1.16MB for the changes in kyle/async-arrow1), so a relatively incompressible increase.

This kind of makes me want to add PR package size reports 🤔 .

Reply:
> This kind of makes me want to add PR package size reports 🤔 .

That sounds awesome!